services/motion/explore/explore.go

```diff
@@ -376,7 +375,7 @@ func (ms *explore) checkForObstacles(
 				worldState,
 				ms.frameSystem,
 				currentPose,
-				currentInputs,
+				plan.Trajectory()[0],
```

Comment:
[q] Why does the CheckPlan function now take in the plan trajectory starting pose instead of the current inputs of the robot? Are currentInputs considered redundant as we also send in currentPose?

Reply:
currentInputs is being stored within planTraj above.  We need this new data type and cannot take currentInputs anymore

Reply:
Ahh I see, I'm used to currentInputs being fed from kienmatic base, but they are in fact the step of the trajectory plan that is currently being handled by the base. Is that correct?

Reply:
Well earlier in this function we assign the beginning of the trajectory to be equal to the output of currentInputs we just need it to be in map form not just an input slice